### PR TITLE
[proto] update protocol version 28 

### PR DESCRIPTION
**Pull request by hexfusion**
This PR adds support for protocol version 28 and AvalancheGo v1.10.9+. It also bumps `protoc-gen-tonic` generated code to v0.3.0 and `protoc-gen-prost` to v0.2.3 and moves them to the community hosted plugins provided by buf. This is a new feature and means that buf does not require the plugins locally to generate code.

The large diff is caused by an update in  `protoc-gen-prost` to support tonic `0.9` which added support for

- max_decoding_message_size
- max_encoding_message_size

This requires all protos to be regenerated.

https://github.com/neoeinstein/protoc-gen-prost/pull/72

### CODE REVIEW

**Comment by hexfusion** [comment]
Not sure what is up with docs and conformance cc @exdx

**Comment by hexfusion** [comment]
Proto p2p updates broke message, will resolve in a bit.

File: crates/avalanche-types/Cargo.toml [hunk 1]
```diff
@@ -94,10 +94,6 @@ prometheus = { version = "0.13.3", default-features = false, features = ["proces
 base64 = { version = "0.21.2", optional = true } # https://github.com/marshallpierce/rust-base64
 num-bigint = { version = "0.4.3", optional = true }
 
-[build-dependencies]
-protoc-gen-prost = "0.2.3"
-protoc-gen-tonic = "0.3.0"
```
Comment (hexfusion) [inline crates/avalanche-types/Cargo.toml:99]: now using community plugins which are hosted by buf

File: crates/avalanche-types/src/message/ping.rs [hunk 2]
```diff
@@ -41,9 +48,9 @@ impl Message {
         let uncompressed_len = encoded.len();
         let compressed = message::compress::pack_gzip(&encoded)?;
         let msg = p2p::Message {
-            message: Some(p2p::message::Message::CompressedGzip(
-                prost::bytes::Bytes::from(compressed),
-            )),
+            message: Some(p2p::message::Message::CompressedGzip(Bytes::from(
+                compressed,
```
Comment (richardpringle) [inline crates/avalanche-types/src/message/ping.rs:52]: FYI, the best times to use `Into` over `From` is when something is being returned or being passed in as an argument.

File: crates/avalanche-types/src/proto/pb/helloworld.tonic.rs
Comment (richardpringle) [inline crates/avalanche-types/src/proto/pb/helloworld.tonic.rs:1]: Why is this here?

File: crates/avalanche-types/src/proto/pb/keystore.tonic.rs
Comment (richardpringle) [inline crates/avalanche-types/src/proto/pb/keystore.tonic.rs:1]: All of these clients should probably be behind feature flags. All this code is probably hurting compile times when it's unlikely that the majority of it is being used.

File: crates/avalanche-types/src/proto/pb/sync.rs [hunk 3]
```diff
@@ -11,133 +12,785 @@ pub mod request {
...
+    0x02, 0x10, 0x0a, 0x0d, 0x0a, 0x05, 0x04, 0x16, 0x02, 0x00, 0x05, 0x12, 0x04, 0xa3, 0x01, 0x02,
+    0x07, 0x0a, 0x0d, 0x0a, 0x05, 0x04, 0x16, 0x02, 0x00, 0x01, 0x12, 0x04, 0xa3, 0x01, 0x08, 0x0b,
+    0x0a, 0x0d, 0x0a, 0x05, 0x04, 0x16, 0x02, 0x00, 0x03, 0x12, 0x04, 0xa3, 0x01, 0x0e, 0x0f, 0x0a,
+    0x0c, 0x0a, 0x04, 0x04, 0x16, 0x02, 0x01, 0x12, 0x04, 0xa4, 0x01, 0x02, 0x12, 0x0a, 0x0d, 0x0a,
+    0x05, 0x04, 0x16, 0x02, 0x01, 0x05, 0x12, 0x04, 0xa4, 0x01, 0x02, 0x07, 0x0a, 0x0d, 0x0a, 0x05,
+    0x04, 0x16, 0x02, 0x01, 0x01, 0x12, 0x04, 0xa4, 0x01, 0x08, 0x0d, 0x0a, 0x0d, 0x0a, 0x05, 0x04,
+    0x16, 0x02, 0x01, 0x03, 0x12, 0x04, 0xa4, 0x01, 0x10, 0x11, 0x62, 0x06, 0x70, 0x72, 0x6f, 0x74,
+    0x6f, 0x33,
 ];
+include!("sync.tonic.rs");
```
Comment (richardpringle) [inline crates/avalanche-types/src/proto/pb/sync.rs:795]: 🤔 .... Does this mean that the whole client implementation is included here?

File: crates/avalanche-types/src/proto/README.md
Comment (richardpringle) [inline crates/avalanche-types/src/proto/README.md:1]: 🥇

File: crates/avalanche-types/src/proto/protos/avalanche/p2p/p2p.proto
Comment (richardpringle) [inline crates/avalanche-types/src/proto/protos/avalanche/p2p/p2p.proto:1]: Why are both the proto-files and the generated code checked into source control? We only need one or the other as code generation is deterministic.

**Review by exdx — APPROVED** [review]
LGTM, thanks

File: crates/avalanche-types/src/proto/pb/helloworld.tonic.rs
Comment (hexfusion) [reply crates/avalanche-types/src/proto/pb/helloworld.tonic.rs:1]: used for testing

File: crates/avalanche-types/src/proto/pb/keystore.tonic.rs
Comment (exdx) [reply crates/avalanche-types/src/proto/pb/keystore.tonic.rs:1]: But we can tackle it in a followup

File: crates/avalanche-types/src/proto/pb/sync.rs (hunk 3 above)
Comment (hexfusion) [reply crates/avalanche-types/src/proto/pb/sync.rs:795]: not the client implementation just the client types.

File: crates/avalanche-types/src/message/ping.rs (hunk 2 above)
Comment (hexfusion) [reply crates/avalanche-types/src/message/ping.rs:52]: Not part of the change but noted.

File: crates/avalanche-types/src/proto/pb/helloworld.tonic.rs
Comment (richardpringle) [reply crates/avalanche-types/src/proto/pb/helloworld.tonic.rs:1]: This needs to be hidden behind `#[cfg(tests)]`

File: crates/avalanche-types/src/proto/pb/sync.rs (hunk 3 above)
Comment (richardpringle) [reply crates/avalanche-types/src/proto/pb/sync.rs:795]: I don't think so. If you look at that file, it's a full-fledged gRPC client implementation for the `sync`-service.

**Review by richardpringle — APPROVED** [review]
Left a "Comment" review as well. I think we need to make some changes, but they are beyond the scope of this PR. The changes here look good.
